nodejs/eks/nodegroup.ts

```diff
@@ -391,59 +414,22 @@ ${customUserData}
 `;
         });
 
+    const version = pulumi.output(args.version || core.cluster.version);
+
+    // https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html
     let amiId: pulumi.Input<string> = args.amiId!;
```

Comment:
I think this type is wrong. This should actually be
```suggestion
    let amiId: pulumi.Input<string> | undefined = args.amiId;
```

Reply:
`imageId`, which consumes `amiId` in the `nodeLaunchConfig`, cannot be undefined

Reply:
The `amiId` arg is defined as
```ts
amiId?: pulumi.Input<string>;
```
which is equivalent to
```ts
amiId: pulumi.Input<string> | undefined
```
https://www.typescriptlang.org/docs/handbook/advanced-types.html#optional-parameters-and-properties

Reply:
If you make the suggested change on 421, I believe that should narrow the type to `pulumi.Input<string>` for you.

Reply:
Done